public/assets/scripts/blocklyflasher.js

```diff
@@ -0,0 +1,1790 @@
+                    }
+                    else
+                    {
+                        listeners[i].call(this);
+                    }
+                }
+            }
+        };
+
+        this.removeListener =  function(type, listener){
+            if (this._listeners[type] instanceof Array){
+                var listeners = this._listeners[type];
+                for (var i=0, len=listeners.length; i < len; i++){
+                    if (listeners[i] === listener){
+                        listeners.splice(i, 1);
+                        break;
+                    }
+                }
+            }
+        }
+    };
+
+    this.setOperationOutput = function(message){
+        $("#cb_cf_operation_output").html(message);
+    };
+
+    this.on = function(type, listener){
+        this.eventManager.addListener(type, listener);
+    };
+
+
+    this.pluginHandler = new function(){
+
+        this.owner;
+
+        this.serialMonitorPort = null;
+
+        this.max_monitor_length = 5000;
+
+        this.uuid4 = function () {
+            var d = new Date().getTime();
+            var uuid = 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function(c) {
+                var r = (d + Math.random()*16)%16 | 0;
+                d = Math.floor(d/16);
+                return (c=='x' ? r : (r&0x7|0x8)).toString(16);
+            });
+            return uuid;
+        };
+
+        this.tabID = this.uuid4();
+
+        this.currentPorts = [];
+
+        this.doflashBootloader = function(programmer, board)
+        {
+            document.getElementById('plugin0').flashBootloader(
+                (this.portslist.selectedIndex == -1 || programmer['communication'] != 'serial')?'':this.portslist.options[this.portslist.selectedIndex].text,
+                programmer['protocol'],
+                programmer['communication'],
+                programmer['speed'],
+                programmer['force'],
+                programmer['delay'],
+                board['bootloader']['high_fuses'],
+                board['bootloader']['low_fuses'],
+                (typeof board['bootloader']['extended_fuses'] === "undefined")?'':board['bootloader']['extended_fuses'],
+                (typeof board['bootloader']['unlock_bits'] === "undefined")?'':board['bootloader']['unlock_bits'],
+                (typeof board['bootloader']['lock_bits'] === "undefined")?'':board['bootloader']['lock_bits'],
+                board['build']['mcu'],
+                bootloader_callback
+            );
+        };
+
+        this.clickedPort = function()
+        {
+            var port = $("#cb_cf_ports").val();
+            var actionId = 43;
+            var metaData = {
+                "selectedPort": port,
+                "tabID": this.tabID
+            };
+            createLogFlasher(actionId, metaData);
+        };
+
+        this.oldPort = null;
+
+        this.logPorts = function () {
+            var oldPort;
+            if (this.oldPort == null)
```

Comment:
It's good practice (actually very very good practice!) to always include curly brackets even if it is a one liner for `if`, `for`, `switch`, etc. It doesn't cost anything and can save hours of crazy debugging.